llvm/include/llvm/Object/ELFObjectFile.h

```diff
@@ -1142,6 +1193,13 @@ ELFObjectFile<ELFT>::getRela(DataRefImpl Rela) const {
   return *Ret;
 }
 
+template <class ELFT>
```

Comment:
Is there any way this can be called before Crels has been populated? If not then please ignore the rest of the comment.

It looks like getRelocatedSection() will need to be called first.

It looks like all the existing use cases here are safe, but if this is part of the general API then maybe worth some additional asserts and pre-conditions given in comments.

Reply:
Changes to this file are for llvm-objdump support. The use patterns ensure that `Crels` will be populated.

I've added the assert to https://github.com/MaskRay/llvm-project/tree/demo-crel